### PR TITLE
kops-ci: Use storage-based instances.

### PR DESCRIPTION
**Pull request by ameukam**
Use a storage-based instance (NVMe local disk of 474 GB) for the EKS cluster.

### CODE REVIEW

**Comment by ameukam** [comment]
Hopefully it should fix:

```
The node was low on resource: ephemeral-storage. Threshold quantity: 3167092657, available: 1824404Ki. Container test was using 2937300Ki, request is 0, has larger consumption of ephemeral-storage. Container sidecar was using 84Ki, request is 0, has larger consumption of ephemeral-storage. 
```

**Comment by ameukam** [comment]
cc @dims @bryantbiggs @dims

File: infra/aws/terraform/kops-infra-ci/eks.tf [hunk 1]
```diff
@@ -99,7 +99,7 @@ module "eks" {
       }
 
       capacity_type  = "ON_DEMAND"
-      instance_types = ["r7i.2xlarge"]
+      instance_types = ["r6id.2xlarge"]
```
Comment (bryantbiggs) [inline infra/aws/terraform/kops-infra-ci/eks.tf:102]: we need to mount and expose the instance store volumes before they can be utilized (and accounted for by kubelet's calculation of ephemeral storage) - here is how we can do that https://github.com/awslabs/data-on-eks/blob/9cde4cf821d21eefea58f83ade62f0cf476d8826/analytics/terraform/spark-k8s-operator/main.tf#L91-L104

Comment (dims) [reply infra/aws/terraform/kops-infra-ci/eks.tf:102]: interesting! thanks for the tip @bryantbiggs

Comment (ameukam) [reply infra/aws/terraform/kops-infra-ci/eks.tf:102]: @bryantbiggs Added.

Comment (bryantbiggs) [reply infra/aws/terraform/kops-infra-ci/eks.tf:102]: related PR for the curious https://github.com/awslabs/amazon-eks-ami/pull/1171

**Review by bryantbiggs — APPROVED** [review]
LGTM!

**Review by iftachk — APPROVED** [review]

File: infra/aws/terraform/kops-infra-ci/eks.tf [hunk 2]
```diff
@@ -119,6 +119,21 @@ module "eks" {
         }
       }
 
+      # NVMe instance store volumes are automatically enumerated and assigned a device
+      pre_bootstrap_user_data = <<-EOT
+        cat <<-EOF > /etc/profile.d/bootstrap.sh
```
Comment (dims) [inline infra/aws/terraform/kops-infra-ci/eks.tf:124]: is there an extra `-` before EOF?

Comment (dims) [reply infra/aws/terraform/kops-infra-ci/eks.tf:124]: never mind .. found it in other places like https://github.com/aws-ia/terraform-aws-eks-blueprints/blob/d2c87b184d2115bfaf6693e66fa2e0b8b470f4ef/patterns/stateful/main.tf#L176

**Comment by ameukam** [comment]
Ref:
 - https://github.com/kubernetes/k8s.io/issues/5127

**Pull request merged**